src/main/java/carpet/commands/DrawCommand.java

```diff
@@ -144,21 +235,269 @@ private static int drawCircle(ServerCommandSource source, BlockPos pos, double r
+                Block blokc = world.getBlockState(blockpos1).getBlock();
+                world.updateNeighbors(blockpos1, blokc);
+            }
+        }
+
+        Messenger.m(source, "gi Filled " + affected + " blocks");
+
+        return 1;
+    }
+
+    private static int drawConey(ServerCommandSource source, BlockPos pos, int radius, int height, boolean pointup, BlockStateArgument block,
+    Predicate<CachedBlockPosition> replacement) {
+        return (int) drawConey(source, pos, radius, height, block, replacement, pointup, false);
+    }
+
+    private static double drawConey(ServerCommandSource source, BlockPos pos, int radius, int height,BlockStateArgument block,
+            Predicate<CachedBlockPosition> replacement, boolean pointup, boolean solid) {
+        int affected = 0;
+        int offset = 0;
+        if(pointup){
+            offset=height-1;
+        }
+        BlockPos.Mutable mbpos = new BlockPos.Mutable(pos);
+
+        List<BlockPos> list = Lists.newArrayList();
+
+        ServerWorld world = source.getWorld();
+
+        CarpetSettings.impendingFillSkipUpdates = !CarpetSettings.fillUpdates;
+
+        for (int x = -radius; x <= radius; ++x) {
+
+            for (int y = -offset; y <= height-offset-1; ++y) {
+
+                for (int z = -radius; z <= radius; ++z) {
+
+                    if (!(x*x+z*z<=(radius/height*y)*(radius/height*y))) {
+                        continue;
+                    }
+
+                    mbpos.set(pos.getX() + x, pos.getY() + y + offset, pos.getZ() + z);
+                    if (replacement == null || replacement.test(new CachedBlockPosition(world, mbpos, true))) {
+                        BlockEntity tileentity = world.getBlockEntity(mbpos);
+
+                        if (tileentity instanceof Inventory) {
+                            ((Inventory) tileentity).clear();
+                        }
+
+                        if (block.setBlockState(world, mbpos, 2)) {
+                            list.add(mbpos.toImmutable());
+                            ++affected;
+                        }
+                    }
+                }
+            }
+        }
+
+        CarpetSettings.impendingFillSkipUpdates = false;
+
+        if (CarpetSettings.fillUpdates) {
+
+            for (BlockPos blockpos1 : list) {
+                Block blokc = world.getBlockState(blockpos1).getBlock();
+                world.updateNeighbors(blockpos1, blokc);
+            }
+        }
+
+        Messenger.m(source, "gi Filled " + affected + " blocks");
+
+        return 1;
+    }
+
+    private static int drawConex(ServerCommandSource source, BlockPos pos, int radius, int height, boolean pointup, BlockStateArgument block,
+    Predicate<CachedBlockPosition> replacement) {
+        return (int) drawConex(source, pos, radius, height, block, replacement, pointup, false);
+    }
+
+    private static int drawConex(ServerCommandSource source, BlockPos pos, int radius, int height,BlockStateArgument block,
```

Comment:
The Y, X and Z cone should use the same code, just parametrized.

Reply:
Done